rust/datafusion/src/optimizer/utils.rs

```diff
@@ -305,6 +312,7 @@ pub fn expr_sub_expressions(expr: &Expr) -> Result<Vec<Expr>> {
             low.as_ref().to_owned(),
             high.as_ref().to_owned(),
         ]),
+        Expr::InList { expr, .. } => Ok(vec![expr.as_ref().to_owned()]),
```

Comment:
Shouldn't this also include the exprs in `list` as well ?

Reply:
Thanks. I have updated the PR with this.